crates/router/src/core/payments/operations/payment_confirm.rs

```diff
@@ -1045,22 +1045,45 @@ impl<F: Clone + Send + Sync> Domain<F, api::PaymentsRequest, PaymentData<F>> for
         _connector_call_type: &ConnectorCallType,
         business_profile: &domain::Profile,
         key_store: &domain::MerchantKeyStore,
+        authentication_product_ids: &serde_json::Value,
     ) -> CustomResult<(), errors::ApiErrorResponse> {
         if let Some(payment_method) = payment_data.payment_attempt.payment_method {
             if payment_method == storage_enums::PaymentMethod::Card
                 && business_profile.is_click_to_pay_enabled
             {
-                let connector_name = CTP_MASTERCARD; // since the above checks satisfies the connector should be click to pay hence hardcoded the connector name
-                let connector_mca = helpers::get_merchant_connector_account(
-                    state,
-                    &business_profile.merchant_id,
-                    None,
-                    key_store,
-                    business_profile.get_id(),
-                    connector_name,
-                    None,
-                )
-                .await?;
+                let authentication_product_ids: HashMap<
+                    String,
+                    common_utils::id_type::MerchantConnectorAccountId,
+                > = authentication_product_ids
+                    .to_owned()
+                    .parse_value("HashMap<String, MerchantConnectorAccountId>")
+                    .change_context(errors::ApiErrorResponse::InternalServerError)
+                    .attach_printable("Error while parsing authentication product ids")?;
```

Comment:
Can we have this field as a concrete type / newtype on the diesel model instead of using `serde_json::Value`?